### ISSUE TITLE
Tauren hitbox, reach and deadzone test

### ISSUE THREAD

**Issue by Beannaa** [magey/classic-warrior#21]
The "hidden racial" of Taurens increases their hitbox by approximately 2 yards and, supposedly, pushes their melee reach and deadzone by the same amount.

- [x] _What exactly is the hitbox size of a Tauren?_
**The hitbox modifier for male tauren is +2.55 yards and +2.2 yards for female tauren.**
**The total hitbox of a male tauren is 5.55 yards and 5.2 yards for female tauren.**
([**Source**](https://youtu.be/By8Ny6BJwKY))

- [x] _What is the "hidden" increased melee reach of a Tauren?_
**The melee range of a tauren is +1.4 yards (as opposed to the standard +2 yards for any other race) thus the total male tauren melee reach is 6.95 yards.**
_(3yd player sphere + 2.55yd male tauren hitbox modifier + 1.4yd melee range from sphere boundary)_
([**Source**](https://youtu.be/By8Ny6BJwKY), [**Source 2**](https://i.imgur.com/FAR2PQJ.jpg))

- [x] _Does it affect ranged attacks as well?_
**Tauren hunters and casters can use ranged attacks from +2.55/+2.2 yards further.**
_Ranged attacks such as autoshot and ranged targeted spells such as Lightning Bolt are 'edge to edge'. That means the actual range indicated in the tooltip is measured between the edges of two characters' hitboxes, not their center point._
([**Source**](https://youtu.be/By8Ny6BJwKY))

- [x] _Does the increased hitbox affects AoEs?_
**PBAoEs such as Arcane Explosion or Frost Nova do not hit a tauren from further away but use a normalized 3 yards player sphere when calculating range.**
([**Source**](https://youtu.be/By8Ny6BJwKY))

- [x] _How does it affect the deadzone?_
**The tauren deadzone is 6.6 yards (6.95-13.55 yards), slightly bigger than the standard 6 yards deadzone (5-11 yards).**
([**Source**](https://youtu.be/By8Ny6BJwKY))

- [x] _Do Taurens benefit more from the infamous "melee leeway"?_
**Taurens gain +2.66 yards from melee leeway, like any other race.**
([**Source 1**](https://youtu.be/S3GH5cWPFOE), [**Source 2**](https://i.imgur.com/qVNF1D9.jpg), [**Source 3**](https://i.imgur.com/FAR2PQJ.jpg))

- [ ] Does it increase the maximum range for Charge and Intercept? _(Duel against a different race)_
- [ ] Does it affect PBAoE skills' range? _(test with Whirlwind)_

**Comment by watchyoursixx** [comment]
Dead zone is currently 5-11 yds on the beta for normal races, btw. If you believe that to be wrong like me pls report <3

Off the top of my head without searching for proof... It does not have any effect on WW or PBAoE attacks from what I've seen on the beta of sonydigital. I might try to find a clip of what I'm referring to or someone else can.

And Yes Tauren gets leeway. https://clips.twitch.tv/ZanySucculentBobaFUNgineer notice he is able to melee then move about a yd to charge.

Confirmation the dead zone is 11 yds on beta: https://www.twitch.tv/karmond/clip/CorrectBrightHippoCopyThis
That's for hunters, but shows that the range calc for non-melee abilities includes the 3 yds from combat reach, so while it should be compared to charge for verification, it is 11 yds on the beta for the dead zone. This also means the actual max range of abilities is 3 yds extended. Pretty hard to verify that though as the only range checks in the API is 9.9 yds, 11.11 yds, and 28 yds afaik.

Tauren with leeway would have approx 7+2.66=9.66 yd melee range vs 11 yd charge range. This lines up with the range difference you see in the sony clip, though it doesn't explain why hunter auto-shot range gets shifted by the extended range i.e. its like 14-15 yds min shoot range as shown in the perp clip from a while back.

**Comment by Beannaa** [comment]
According to [this video](https://youtu.be/By8Ny6BJwKY) the hitbox modifier for male tauren is +2.55 yards and +2.2 yards for female tauren.
Which brings the total hitbox of a male tauren to 5.55 yards and female to 5.2 yards.

The standard melee range is +2 yards but for tauren it seems to be only +1.4 yards, which brings the total male tauren melee reach to 6.95 yards or pretty much 7yd. (3yd player sphere + 2.55yd male tauren hitbox modifier + 1.4yd melee range from sphere boundary)
The melee reach of any other race is the standard +2 yards melee range added to the standard 3 yards player sphere, for a total of 5 yards.

The deadzone seems to be pushed as well. According to the video, the standard deadzone is 11 yards (8yd minimal distance + 3yd player sphere) while it's 13.55 yards for a tauren due to his larger player sphere.

This video also shows that PBAoEs do not take the hitbox modifier of an enemy player into account. The exemple shows that, if an Arcane Explosion does not hit a standard hitbox (3yd) right outside out of the 10 yards range (10.2yd in the exemple), it still won't hit a tauren at the same position, despite his hitbox technically now inside the 10 yards range when factoring the additional 2.55 yards.
We can assume the same applies to Frost Nova and maybe Freezing Trap as well, which means a tauren player would not be penalized for his hitbox when dealing with those skills.

While we can assume that Warrior's PBAoEs such as Whilwind do not benefit from the tauren's increased hitbox since the center of the PBAoE should still be the center of the hitbox, this video does not prove it.

**Comment by oNodrak** [comment]
Apologies in advance for the long post, but this has wide ranging implications.

>     * [x]  _Does the increased hitbox affects AoEs?_
>       **PBAoEs such as Arcane Explosion or Frost Nova do not hit a tauren from further away but use a normalized 3 yards player sphere when calculating range.**
>       ([**Source**](https://youtu.be/By8Ny6BJwKY))

This right here is evidence of the issue:  https://youtu.be/By8Ny6BJwKY?t=91
10.2 yards, center to center, and 10 yard AE does not hit.

As per 2014: https://gyazo.com/d32ae93db548aae28ce30e8487aeb9a4 it should hit the tauren, but it does not.

What is happening:

In 2.3 blizzard changed hunters from 8-35 range defined in a table, to 0-35 range.  At the same time they changed the way Ranged Attacks target things.  Charge in the same table is 8-35 before and after.  Why did they change ranged?  Because in order for 0 min range to 'work', it has to function with the same target rules as melee (what I call "edge to edge").  Previously ranged worked as ("Center to Edge").  It also has to be 'overruled' by melee as well.

This can be derived from the same combat table range.  'Melee' range is listed as 0-5.  However, this was the only 'range' with a special flag in 1.X.  This flag is what sets the range to extend from the edge of the player hitbox ("edge instead of center").  Come 2.3 patch change, and hunter range now has its own special flag as well.  This flag is what should prevent 4 yard auto shots.

An astute observer might have noticed that there was never any talk about 'Tauren Spell Range' or 'Tauren Hunter Range', the only mentions were 'melee range'.  This is because to my knowledge all spells ignore the caster radius ("Center to Edge").  All hunter shots were spells with a special range flag and damage flag, "Ranged Spell with Type Melee", which gives hunters their unique attack system rules.  (Paladin spell attacks are the same flag, with a different range flag).

Now come Modern WoW.  Blizzard is homogenizing and refactoring.  At this point, I stopped caring in pvp and pve, so I cannot tell you if spells changed from "Center to Edge" to "Edge to Edge" or not, but that should be quick to test on live if someone does care.   

Now come Classic WoW.  Blizzard has ported the data tables, but not the systems.  This should be clearly evident now.  This means the Hunter Range table row is back to 8-35, but also the special flag is gone.  Now with 8-35 range, the flag is not really needed, but the system is still the 2.3+ system.  Hunter attacks are still "Edge to Edge".  

What does it all mean:  

Melee should be Edge-Edge*  
Charge should be Center-Edge
Spells should be Center-Edge  
Autoshot should be Center-Edge  
AoE Spells should be Center-Center
Autoshot is Edge-Edge
AoE Spells are Center-Center


_*The Caveat here is that it is still measured Center-Center in terms of 0-X, the Tauren modifier provides the effect of Edge-Edge_

Therefore you should be able to charge a hunter inside their deadzone as a test.  If there is no distinction between Charge and Autoshot, then All Spells are using modern wow ranges.

We can use the incorrect blue post as evidence of these changes.  PointBasedAoE never took target radius into account because it would be heavy on the server.  (See Frostnova+Blastwave dead-zone kiting needing to step foward).

Furthermore, at the end of the video, 5 yard melee range is not long enough to attack past a 2 yard trap radius (3+2=5), if the Tauren radius applied more universally (circle size change, instead of special rules), this would also trigger the trap.  Ergo, Tauren modifiers are direct to 'melee range' and are being erroneously applied to other situations (Ranged Attacks from Hunters).  If/Because this is the case, we can also test if all spells are targeting incorrectly as per above.

Naturally if I were in the beta, I could test this and put it to rest, but alas...

**TL:DR**:https://docs.google.com/spreadsheets/d/1TeY2oJdOWfkBA0c122wfgBD04c7Drt7IMACNhoCaF4o/edit?usp=sharing

I have  been tweaking and adding to the sheet above to hopfully add some clarity.  The Charge video in the Venruki duel above aligns with ("Center to Edge") targeting for Charge.

**Comment by oNodrak** [comment]
Further evidence that Hunter Attacks are using 2.3+ systems, their shots cannot be blocked currently.  
https://us.forums.blizzard.com/en/wow/t/bug-ranged-attacks-cannot-be-blocked/207452

**Comment by watchyoursixx** [comment]
Just as an update, based on this [blue post](https://us.forums.blizzard.com/en/wow/t/hunter-beta-issues-update/223166) it's now been confirmed that the dead zone for hunters (and in turn, charge dead zone) is the same as 1.12. With this knowledge, I'd say its safe to assume the blue from 2014 is accurate for classic as well, though with slight variations in the "1.5 yd reach" number based on female/male/tauren/nontauren.

The whole "3 yd player sphere" thing has always thrown me off, as that blue from 2014 listed the melee range calculation as MAX(playercombatreach + targetcombatreach + 4/3, 5), with player/target most often being 1.5 yds. This seems to make the most sense to me and implies the 4/3 is a base hitbox, not anything else. Any additional yardage to a hitbox would be considered "reach".

I could be wrong and they formula could be different, but based on the others being correct I don't see why the melee one would be off. 

The range of all non-melee abilities should be increased by a minimum of 3 yds. Effectively you would never notice it as everything is affected by it, but it basically means all of the tooltip ranges are off by 3 yards for non-melee.

**Comment by oNodrak** [comment]
The 3-yard concept is just a simplification for the back-end math.  It makes the function faster in time-scaling if you assume the size of the target based on already known information.  This is why PBAoE is Center-Center.  The 4/3 factor is/should only applied in the case of taurens/npcs, which I find extra odd.  

The only problem I have with the combat reach formula is that it doesn't fit any known numbers.  The game files have a value of (1.35 and 1.25 for Tauren M and F, and 1 for all other non-gnome).  NPCs have additional values that I believe indicate 'Attack Range, and Hitbox Size'.  

Eg, Onyxia is Size 4, with standard Range, while Ragnaros is Size 4 with Range 4.

To check the player range we have to use taurens:

(1.5+1.35 + 1.5+1.35 + 4/3) = 7.03 (expected 9.5)  
(1.5*1.35 + 1.5*1.35 + 4/3) = 5.38  
(1.5 + 1.5 + 4/3) * 1.35 * 1.35 = 7.89  
Max(1.5 + 1.5 + 4/3 , 5) * 1.35 * 1.35 = 9.11  

If we solve:
9.5 = (2x + 4/3), x = 4.08 male
8.8 = (2y + 4/3), y= 3.73 female  
All other races would be z= 2.50 [since they all trigger Max(5)].

But lets check Human v Tauren:  

Max(1.5 + 4.08 + 4/3, 5) = 6.91 (expected 6.95)   
Max(1.5 + 3.73 + 4/3, 5) = 6.57 (expected 6.60)   

The dead-zone being confirmed does not exactly surprise me, it felt accurate on the non-taurens, and Tauren hunter was probably a tiny demographic.  Anyways, we can at least confirm that the formula may be in effect for player vs player combat, even if the range values are nonsense and waste cpu cycles on most combat.  My contention is that the Hunter Attacks should be C-E instead of E-E, and I don't think that got tested.
